homeassistant/components/home_connect/select.py

```diff
@@ -471,10 +473,13 @@ async def async_added_to_hass(self) -> None:
                 )
 
         if setting and setting.constraints and setting.constraints.allowed_values:
+            self._original_option_keys = set(setting.constraints.allowed_values)
             self._attr_options = [
-                self.entity_description.values_translation_key[option]
-                for option in setting.constraints.allowed_values
-                if option in self.entity_description.values_translation_key
+                self.entity_description.values_translation_key.get(
+                    option, bsh_key_to_translation_key(option)
```

Comment:
Why do we need the default? To avoid breaking the integration it would be enough to use `dict.get` and not include the options that are `None`.

We said before that we won't include unknown options. I think we should continue with that line.